src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -169,224 +176,668 @@ private sealed class ColInfoEx
+            {
                 _ldas[i] = new LdaState(Host, ctx);
                 _exes[i] = _ldas[i].InfoEx;
                 _types[i] = new VectorType(NumberType.Float, _ldas[i].InfoEx.NumTopic);
             }
-            using (var ent = ctx.Repository.OpenEntryOrNull("model", WordTopicModelFilename))
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (_ldas != null)
+            {
+                foreach (var state in _ldas)
+                    state?.Dispose();
+            }
+            if (disposing)
+                GC.SuppressFinalize(this);
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        ~LdaTransform()
+        {
+            Dispose(false);
+        }
+
+        public LdaState GetLdaState(int iinfo)
+        {
+            Contracts.Assert(0 <= iinfo && iinfo < _ldas.Length);
+            return _ldas[iinfo];
+        }
+
+        // Factory method for SignatureLoadDataTransform.
+        private static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+            => Create(env, ctx).MakeDataTransform(input);
+
+        // Factory method for SignatureLoadRowMapper.
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        // Factory method for SignatureDataTransform.
+        public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+            env.CheckValue(input, nameof(input));
+
+            env.CheckValue(args.Column, nameof(args.Column));
+            var cols = new ColumnInfo[args.Column.Length];
+            using (var ch = env.Start("ValidateArgs"))
             {
-                _saveText = ent != null;
+
+                for (int i = 0; i < cols.Length; i++)
+                {
+                    var item = args.Column[i];
+                    cols[i] = new ColumnInfo(item.Source,
+                        item.Name ?? item.Source,
+                        item.NumTopic ?? args.NumTopic,
+                        item.AlphaSum ?? args.AlphaSum,
+                        item.Beta ?? args.Beta,
+                        item.Mhstep ?? args.Mhstep,
+                        item.NumIterations ?? args.NumIterations,
+                        item.LikelihoodInterval ?? args.LikelihoodInterval,
+                        item.NumThreads ?? args.NumThreads ?? 0,
+                        item.NumMaxDocToken ?? args.NumMaxDocToken,
+                        item.NumSummaryTermPerTopic ?? args.NumSummaryTermPerTopic,
+                        item.NumBurninIterations ?? args.NumBurninIterations,
+                        item.ResetRandomGenerator ?? args.ResetRandomGenerator);
+                };
```

Comment:
>}; [](start = 16, length = 2)

remove the ;

Reply:
fixed

---
In reply to: [231645863](https://github.com/dotnet/machinelearning/pull/1410#discussion_r231645863) [](ancestors = 231645863)